spyder/config/main.py

```diff
@@ -494,8 +499,90 @@
               'advanced/port': 2087,
               'advanced/external': False,
               'advanced/stdio': False
-             })
+             }),
+            ]
+
+
+NAMEMAP = {
+    # Empty container object means use the rest of defaults
+    'main': {},
+    # Splitting these files makes sense for projects, we might as well
+    # apply the same split for the app global config
+    'lsp': [('lsp-server', [])],
+    # These options change on spyder startup or are tied to a specific OS,
+    # not good for version control
+    'transient': [
+        ('main', [
+            'completion/size',
+            'crash',
+            'current_version',
+            'historylog_filename',
+            'last_visible_toolbars',
+            'spyder_pythonpath',
+            'window/position',
+            'window/prefs_dialog_size',
+            'window/size',
+            'window/state',
             ]
+         ),
+        ('appearance', [
+            'windows_style',
+            ]
+         ),
+        ('editor', [
+            'autosave_mapping',
+            'bookmarks',
+            'filenames',
+            'layout_settings',
+            'recent_files',
+            'splitter_state',
+            ]
+         ),
+        ('explorer', [
+            'file_associations',

```

Comment:
Well, I included things here that either:
Are transient or are OS specific (file associations will be different for each OS so that was why)

But if we limit to just files that change a lot, then I can revert this

Reply:
@ccordoba12 thoughts on this?

Reply:
The problem is this: what if you share several computers with the same OS and the same (ot mostly the same) programs? That could be very well the case in a corporate environment.

So let's discuss this in our weekly TDK meeting.